tests/test_dataflow/core/test_lispress.py

```diff
@@ -234,3 +273,8 @@ def test_type_args_in_program():
     assert len(program.expressions) == 1
     assert program.expressions[0].type_args == [TypeName("PleasantryCalendar", ())]
     assert program.expressions[0].type is None
+
+
+def test_fully_typed_reference():
+    s = "(lambda (^Unit x0) ^Unit x0)"
+    assert round_trip_through_program(s) == "(lambda (^Unit x0) x0)"
```

Comment:
Not worth worrying about, but maybe a comment about why we lose the type ascription?

Reply:
this is really to test that it doesn't throw an exception. i'll make the test more specific and add a comment